numpy/core/src/multiarray/calculation.c

```diff
@@ -86,6 +97,24 @@ PyArray_ArgMax(PyArrayObject *op, int axis, PyArrayObject *out)
     if (ap == NULL) {
         return NULL;
     }
+
+    // Decides the shape of the output array.
+    if (!keepdims) {
+        out_ndim = PyArray_NDIM(ap) - 1;
+        out_shape = PyArray_DIMS(ap);
+    } else {
+        out_shape = _shape_buf;
+        if (axis_copy == NPY_MAXDIMS) {
+            for( int i = 0; i < out_ndim; i++ ) {
+                out_shape[i] = 1;
+            }
+        } else {
+            out_ndim = PyArray_NDIM(ap);
+            memcpy(out_shape, PyArray_DIMS(ap), out_ndim * sizeof(npy_intp));
+            out_shape[out_ndim - 1] = 1;
+        }
```

Comment:
```C
npy_intp *original_op_shape = PyArray_DIMS(op);
Doesn't this work, and you can just delete everything at the end of the function?

```
<...>
```suggestion
            memcpy(out_shape, original_op_shape, out_ndim * sizeof(npy_intp));
            out_shape[axis] = 1;
```
With an additional comment, maybe something like:
```C
/* 
 * While `ap` may be transposed, we can ignore this for `out` because it the
 * transpose only reorders the size 1 `axis` (not changing memory layout).
 */
```

Reply:
I will try this suggestion. If it works, I will push the changes here.

Reply:
Done. Thanks. Let me know if there is something more to do.